test/adaptive_load/scoring_function_test.cc

```diff
@@ -0,0 +1,204 @@
+  config.mutable_upper_threshold()->set_value(5.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(4.0), 1.0);
+}
+
+TEST(BinaryScoringFunction, EvaluateMetriceturnsOneForValueEqualToUpperThreshold) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_upper_threshold()->set_value(5.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(5.0), 1.0);
+}
+
+TEST(BinaryScoringFunction, EvaluateMetricReturnsNegativeOneForValueOutsideUpperThreshold) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_upper_threshold()->set_value(5.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(6.0), -1.0);
+}
+
+TEST(BinaryScoringFunction, EvaluateMetricReturnsOneForValueWithinLowerThreshold) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_lower_threshold()->set_value(5.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(6.0), 1.0);
+}
+
+TEST(BinaryScoringFunction, EvaluateMetricReturnsOneForValueEqualToLowerThreshold) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_lower_threshold()->set_value(5.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(5.0), 1.0);
+}
+
+TEST(BinaryScoringFunction, EvaluateMetricReturnsNegativeOneForValueOutsideLowerThreshold) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_lower_threshold()->set_value(5.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(4.0), -1.0);
+}
+
+TEST(BinaryScoringFunction, EvaluateMetricReturnsOneForValueWithinThresholdRange) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_lower_threshold()->set_value(5.0);
+  config.mutable_upper_threshold()->set_value(7.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(6.0), 1.0);
+}
+
+TEST(BinaryScoringFunction, EvaluateMetricReturnsOneForValueEqualToLowerThresholdOfRange) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_lower_threshold()->set_value(5.0);
+  config.mutable_upper_threshold()->set_value(7.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(5.0), 1.0);
+}
+
+TEST(BinaryScoringFunction, EvaluateMetricReturnsOneForValueEqualToUpperThresholdOfRange) {
+  nighthawk::adaptive_load::BinaryScoringFunctionConfig config;
+  config.mutable_lower_threshold()->set_value(5.0);
+  config.mutable_upper_threshold()->set_value(7.0);
+  BinaryScoringFunction scoring_function(config);
+  EXPECT_EQ(scoring_function.EvaluateMetric(7.0), 1.0);
+}
+
+TEST(BinaryScoringFunction, EvaluateMetricReturnsNegativeOneForValueBelowThresholdRange) {
```

Comment:
One last remark: there's a fair bit of repetitiveness here, which could be condensed via `TEST_P`.
We'd loose the helpful specificity of the test names, which indicate what is being tested. Maybe a hybrid of `TEST_P` and named constants could provide a middle ground?

Reply:
Done, it's much simpler with TEST_P.